src/btree/bt_page.c

```diff
@@ -1075,8 +1075,11 @@ __inmem_row_leaf(WT_SESSION_IMPL *session, WT_PAGE *page, bool *preparedp)
              *
              * The visibility information is not referenced on the page so we need to ensure that
              * the value is globally visible at the point in time where we read the page into cache.
+             * Pages from checkpoint-related files that have been pushed onto the pre-fetch queue
+             * will be comprised of data that is globally visible, and so the reader thread which
+             * attempts to populate the page into cache can skip the visible all check.
```

Comment:
NP: I like this comment! Just one thing - could we say "read the page into cache" here for more clarity?

Reply:
Fixed in [3a516b7](https://github.com/wiredtiger/wiredtiger/pull/9925/commits/3a516b74fa137c3c5dff4526234cc7c5cdd5764a).